Application.mk

```diff
@@ -191,6 +192,21 @@ else
 
 MAINNAME := $(addsuffix _main,$(PROGNAME))
 
+ifeq ($(CONFIG_WINDOWS_NATIVE),y)
+$(MAINCXXOBJ): %$(CXXEXT)$(SUFFIX)$(OBJEXT): %$(CXXEXT)
+	$(eval $<_CXXFLAGS += ${shell $(DEFINE) "$(CXX)" main -val $(firstword $(MAINNAME))})
```

Comment:
because define.bat is different from the define.sh, and usage is also different this is usage of define.bat:
`echo USAGE:%progname% [-h] ^<compiler-path^> [-val ^<^val1^>] [^<def2^> [-val ^<val2^>] [^<def3^> [-val ^<val3^>] ...]]
echo Where:"
echo  ^<compiler-path^>
echo    The full path to your ccpath
echo  ^<def1^> ^<def2^> ^<def3^> ...
echo    A list of pre-preprocesser variable names to be defined.
echo  [-val ^<val1^>] [-val ^<val2^>] [-val ^<val3^>] ...
echo    optional values to be assigned to each pre-processor variable.
echo    If not supplied, the variable will be defined with no explicit value.`

without -val, there will fatal error in Windows native build, for example:
Dnsh_main become Dnsh Dmain

Reply:
@zouboan it's betteer to sync define.bat and define.sh usage to avoid the duplication here.

Reply:
is a good idea! although i'm not expert in .bat and .sh but I wouldn't mind having a try

Reply:
Ok, let's wait your result.